src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -78,6 +79,126 @@ static NSStringCompareOptions ConvertFromCompareOptionsToNSStringCompareOptions(
+
+    return modifiedString;
+}
+
+// Structure to hold the modified string and character mapping
+typedef struct {
+    NSString* modifiedString;
+    NSArray<NSNumber*>* mapping; // mapping[i] = original position j for character at position i in modifiedString
+} StringWithMapping;
+
+/**
+ * Removes symbols including whitespace, punctuation, currency symbols, the percent sign, 
+ * mathematical symbols, the ampersand, and similar characters from the specified string.
+ * This corresponds to .NET's CompareOptions.IgnoreSymbols behavior.
+ */
+static NSString* RemoveSymbols(NSString* source)
+{
+    NSError *error = nil;
+    // Unicode categories covered:
+    // \p{P} - Punctuation (all punctuation marks)
+    // \p{S} - Symbols (currency, mathematical, modifier, and other symbols)
+    // \p{Z} - Separators (space, line, and paragraph separators)
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"[\\p{P}\\p{S}\\p{Z}]" options:NSRegularExpressionCaseInsensitive error:&error];
+
+    if (error != nil)
+        return source;
+
+    NSString *modifiedString = [regex stringByReplacingMatchesInString:source options:0 range:NSMakeRange(0, [source length]) withTemplate:@""];
+
+    return modifiedString;
+}
+
+/**
+ * Removes symbols including whitespace, punctuation, currency symbols, the percent sign, 
+ * mathematical symbols, the ampersand, and similar characters from the specified string.
+ * Returns both the modified string and a mapping array where mapping[i] contains the original
+ * position in the source string for the character at position i in the modified string.
+ * This corresponds to .NET's CompareOptions.IgnoreSymbols behavior.
+ */
+static StringWithMapping RemoveSymbolsWithMapping(NSString* source)
+{
+    StringWithMapping result = {source, nil};
+    
+    NSError *error = nil;
+    // Unicode categories covered:
+    // \p{P} - Punctuation (all punctuation marks)
+    // \p{S} - Symbols (currency, mathematical, modifier, and other symbols)
+    // \p{Z} - Separators (space, line, and paragraph separators)
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"[\\p{P}\\p{S}\\p{Z}]" options:NSRegularExpressionCaseInsensitive error:&error];
+
+    if (error != nil)
+        return result;
+
+    NSMutableString *modifiedString = [NSMutableString string];
+    NSMutableArray<NSNumber*> *mapping = [NSMutableArray array];
+    
+    NSUInteger sourceLength = [source length];
+    for (NSUInteger i = 0; i < sourceLength; i++)
+    {
+        unichar ch = [source characterAtIndex:i];
+        NSString *charString = [NSString stringWithCharacters:&ch length:1];
+        
+        // Check if this character matches the symbol pattern
+        NSRange matchRange = [regex rangeOfFirstMatchInString:charString options:0 range:NSMakeRange(0, 1)];
```

Comment:
>NSRange matchRange = [regex rangeOfFirstMatchInString:charString options:0 range:NSMakeRange(0, 1)]; [](http://example.com/codeflow?start=8&length=100)

I am not sure how the performance will look like with that? Can't this be done in the managed code using the CharUnicodeInfo?